lib/manager/components/LabelAssigner.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import React from 'react'
+import { connect } from 'react-redux'
+
+import { updateFeedSource } from '../actions/feeds'
+import FeedLabel from '../../common/components/FeedLabel'
+import { stringifyLabels } from '../util'
+import type { Feed, Project } from '../../types'
+
+type Props = {
+ feedSource: Feed,
+ project: Project,
+ updateFeedSource: Function
+}
+
+class LabelAssigner extends React.Component<Props> {
+  onLabelClick = (labelId: String) => {
+    const {feedSource} = this.props
+    const updatedLabels = stringifyLabels(feedSource.labels)
+
+    const labelIndex = updatedLabels.indexOf(labelId)
+
+    // Either remove the label from the list, or add it
+    if (labelIndex < 0) {
+      updatedLabels.push(labelId)
+    } else {
+      updatedLabels.splice(labelIndex, 1)
+    }
+
+    this.props.updateFeedSource(feedSource, { labels: updatedLabels })
+  }
+
+  render () {
+    const { project, feedSource } = this.props
+    const projectLabels = project.labels
+    const fsLabelIds = feedSource.labels.map((label) => label.id)
+
+    return (
+      <div className='feedLabelContainer large'>
```

Comment:
It could be, but as far as I can tell `<React.Select>` doesn't let you see all labels/options at once! Is that something we're willing to give up? My upcoming commit moves the checkboxes outside the label, and I'm hoping that meets our needs.